benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/AzureMatcherBenchmark.cs

```diff
@@ -28,12 +31,54 @@ public void Setup()
             // of the request data.
             _samples = SampleRequests(EndpointCount, SampleCount);
 
-            _route = SetupMatcher(RouteMatcher.CreateBuilder());
-            _tree = SetupMatcher(TreeRouterMatcher.CreateBuilder());
+            _baseline = (BarebonesMatcher)SetupMatcher(new BarebonesMatcherBuilder());
+            _dfa = SetupMatcher(new DfaMatcherBuilder());
+            _instruction = SetupMatcher(new InstructionMatcherBuilder());
+            _route = SetupMatcher(new RouteMatcherBuilder());
+            _tree = SetupMatcher(new TreeRouterMatcherBuilder());
 
             _feature = new EndpointFeature();
         }
 
+        [Benchmark(Baseline = true, OperationsPerInvoke = SampleCount)]
+        public async Task Baseline()
+        {
+            var feature = _feature;
+            for (var i = 0; i < SampleCount; i++)
+            {
+                var sample = _samples[i];
+                var httpContext = _requests[sample];
+                await _baseline._matchers[sample].MatchAsync(httpContext, feature);
```

Comment:
Note that the barebones matcher 'cheats'. We always know which template to use when testing against the barebones matcher. This is good because it sets a lower bound 👍